### PR TITLE
E2E Tests: Disable access to remote block directory and patterns

### PR DESCRIPTION
**Pull request by swissspidy**
## Context

They often cause block validation console errors on older WordPress version. This currently causes our e2e tests against WP 5.9 to fail.

## Summary

Disable access to remote block directory and patterns

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #

### CODE REVIEW

File: packages/e2e-tests/src/plugins/disable-block-directory.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\E2E\BlockDirectory;
+
+add_action(
+	'admin_init',
+	function() {
```
Comment (spacedmonkey) [inline packages/e2e-tests/src/plugins/disable-block-directory.php:35]: Could this be a arrow function?

File: bin/local-env/install-wordpress.sh [hunk 2]
```diff
@@ -194,7 +197,7 @@ fi
 
 MEDIA_TRASH_CURRENT=!MEDIA_TRASH;
 if [[ "$(wp config has --type=constant MEDIA_TRASH)" ]]; then
-  $MEDIA_TRASH_CURRENT=$(wp config get --type=constant --format=json MEDIA_TRASH | tr -d '\r')
+  MEDIA_TRASH_CURRENT=$(wp config get --type=constant --format=json MEDIA_TRASH | tr -d '\r')
```
Comment (spacedmonkey) [inline bin/local-env/install-wordpress.sh:200]: Is this change related?

Comment (swissspidy) [reply bin/local-env/install-wordpress.sh:200]: Not really, but fixed it while at it.

File: packages/e2e-tests/src/plugins/disable-block-directory.php (hunk 1 above)
Comment (swissspidy) [reply packages/e2e-tests/src/plugins/disable-block-directory.php:35]: ~Yes definitely.~
Edit: no, because it's a multi-line function

File: bin/local-env/install-wordpress.sh [hunk 3]
```diff
@@ -160,6 +160,9 @@ wp plugin install classic-widgets --force --quiet
 echo -e $(status_message "Activating Twenty Twenty theme...")
 wp theme install twentytwenty --activate --force --quiet
 
+echo -e $(status_message "Disable block directory and remote patterns...")
+wp plugin activate e2e-tests-disable-block-directory --quiet
```
Comment (spacedmonkey) [inline bin/local-env/install-wordpress.sh]: Instead of this plugin, I wonder if we could use a plugin like [airplane-mode](https://github.com/norcross/airplane-mode).

Comment (swissspidy) [reply bin/local-env/install-wordpress.sh]: I thought about it as well but it doesn't appear to handle this specific functionality. But it does way more things, so I'd be worried about anything else breaking because of it.

Comment (spacedmonkey) [reply bin/local-env/install-wordpress.sh]: Isn't that a good thing, it doesnt call to the internet, might make the tests faster.

Comment (spacedmonkey) [reply bin/local-env/install-wordpress.sh]: Does it make sense to activate the plugin here. 

https://github.com/GoogleForCreators/web-stories-wp/blob/f2e523ef807e028475b9467b2441fe0069cd67df/packages/e2e-tests/src/config/bootstrap.js#L251

Comment (swissspidy) [reply bin/local-env/install-wordpress.sh]: > Does it make sense to activate the plugin here.

`bootstrap.js` is useful for doing things between runs, like deleting media and trashing posts. But in my eyes this is more a once-and-forget-it thing for the environment. Not something that you deactivate later. Just like we activate the RTL Tester once and keep it active.

> Isn't that a good thing, it doesnt call to the internet, might make the tests faster.

Sure, but it would be something _in addition_ to this plugin here. We can give it a try.

**Comment by swissspidy** [comment]
@spacedmonkey FYI, removed Airplane Mode again because it caused too many tests to fail, for example around media hotlinking or when using the AMP CDN (Web Stories block, AMP validator)

File: includes/namespace.php [hunk 4]
```diff
@@ -41,7 +41,7 @@
  *
  * @param bool $network_wide Whether to activate network-wide.
  */
-function activate( bool $network_wide = false ): void {
+function activate( ?bool $network_wide = false ): void {
```
Comment (swissspidy) [inline includes/namespace.php:44]: Changed this because it's not always a bool, e.g. when activating a plugin via WP-CLI.

See https://github.com/wp-cli/extension-command/issues/348

File: packages/e2e-tests/src/jest.config.js [hunk 5]
```diff
@@ -46,8 +46,8 @@ export default {
   setupFilesAfterEnv: [
     'jest-extended/all',
     '<rootDir>/packages/e2e-tests/src/config/bootstrap.js',
-    '@wordpress/jest-puppeteer-axe',
-    '@wordpress/jest-console',
+    '<rootDir>/packages/e2e-tests/node_modules/@wordpress/jest-puppeteer-axe',
+    '<rootDir>/packages/e2e-tests/node_modules/@wordpress/jest-console',
```
Comment (swissspidy) [inline packages/e2e-tests/src/jest.config.js:50]: Since these packages are not always in the `node_modules` folder in the root folder (because of npm workspaces), this is the way.

**Comment by swissspidy** [comment]
Those timeouts are not great 😕  No idea why tests suddenly take that long...

**Comment by swissspidy** [comment]
Timeout is not resolved yet, but merging to unblock other PRs.
